test/e2e/serial/tests/configuration.go

```diff
@@ -736,9 +737,105 @@ var _ = Describe("[serial][disruptive][slow] numaresources configuration managem
+			Expect(err).ToNot(HaveOccurred(), "cannot get %q in the cluster", dsKey.String())
+			expectedTolerations := rteManifests.DaemonSet.Spec.Template.Spec.Tolerations // shortcut
+			gotTolerations := dsObj.Spec.Template.Spec.Tolerations                       // shortcut
+			expectEqualTolerations(gotTolerations, expectedTolerations)
+
+			By("adding extra tolerations")
+			updatedNropObj := setRTETolerations(context.TODO(), fxt.Client, nroKey, []corev1.Toleration{sriovToleration()})
+			By("waiting for DaemonSet to be ready")
+			_, err = wait.With(fxt.Client).Interval(10*time.Second).Timeout(1*time.Minute).ForDaemonSetUpdateByKey(context.TODO(), dsKey)
+			Expect(err).ToNot(HaveOccurred(), "daemonset %s did not start updated: %v", dsKey.String(), err)
+			_, err = wait.With(fxt.Client).Interval(10*time.Second).Timeout(3*time.Minute).ForDaemonSetReadyByKey(context.TODO(), dsKey)
+			Expect(err).ToNot(HaveOccurred(), "failed to get the daemonset %s: %v", dsKey.String(), err)
+
+			defer func(ctx context.Context) {
+				By("removing extra tolerations")
+				_ = setRTETolerations(ctx, fxt.Client, nroKey, []corev1.Toleration{})
+				By("waiting for DaemonSet to be ready")
+				_, err = wait.With(fxt.Client).Interval(10*time.Second).Timeout(1*time.Minute).ForDaemonSetUpdateByKey(ctx, dsKey)
+				Expect(err).ToNot(HaveOccurred(), "daemonset %s did not start updated: %v", dsKey.String(), err)
+				_, err = wait.With(fxt.Client).Interval(10*time.Second).Timeout(3*time.Minute).ForDaemonSetReadyByKey(ctx, dsKey)
+				Expect(err).ToNot(HaveOccurred(), "failed to get the daemonset %s: %v", dsKey.String(), err)
+			}(context.TODO())
+
+			By("checking the tolerations in the owned DaemonSet")
+			err = fxt.Client.Get(context.TODO(), client.ObjectKey{Namespace: dsKey.Namespace, Name: dsKey.Name}, &dsObj)
+			Expect(err).ToNot(HaveOccurred(), "cannot get %q in the cluster", dsKey.String())
+
+			expectedTolerations = updatedNropObj.Spec.NodeGroups[0].Config.Tolerations // shortcut
+			gotTolerations = dsObj.Spec.Template.Spec.Tolerations                      // shortcut
+			expectEqualTolerations(gotTolerations, expectedTolerations)
+		})
 	})
 })
 
+func expectEqualTolerations(tolsA, tolsB []corev1.Toleration) {
+	GinkgoHelper()
+
+	// TODO: sort, then check
+	Expect(tolsA).To(Equal(tolsB), "mismatched tolerations")
+}
+
+func setRTETolerations(ctx context.Context, cli client.Client, nroKey client.ObjectKey, tols []corev1.Toleration) *nropv1.NUMAResourcesOperator {
+	GinkgoHelper()
+
+	nropOperObj := nropv1.NUMAResourcesOperator{}
+	Eventually(func(g Gomega) {
+		err := cli.Get(ctx, nroKey, &nropOperObj)
+		g.Expect(err).ToNot(HaveOccurred())
+
+		if nropOperObj.Spec.NodeGroups[0].Config == nil {
+			nropOperObj.Spec.NodeGroups[0].Config = &nropv1.NodeGroupConfig{}
+		}
+		nropOperObj.Spec.NodeGroups[0].Config.Tolerations = []corev1.Toleration{
+			sriovToleration(),
```

Comment:
should use tols instead of sriovToleration()

Reply:
yes, thanks for catching this!